api/server.go

```diff
@@ -674,6 +676,376 @@ func (s *server) walletsFundSFHandler(jc jape.Context) {
 	})
 }
 
+func (s *server) walletsConstructHandler(jc jape.Context) {
+	cs := s.cm.TipState()
+	if cs.Index.Height >= cs.Network.HardforkV2.RequireHeight {
+		jc.Error(errors.New("v1 transactions are not allowed after the v2 require height"), http.StatusBadRequest)
+	}
+
+	var walletID wallet.ID
+	if err := jc.DecodeParam("id", &walletID); err != nil {
+		return
+	}
+	var wcr WalletConstructRequest
+	if err := jc.Decode(&wcr); err != nil {
+		return
+	}
+
+	if wcr.ChangeAddress == types.VoidAddress {
+		jc.Error(errors.New("change address must be specified"), http.StatusBadRequest)
+		return
+	}
+
+	var siacoinInput types.Currency
+	for i, sco := range wcr.Siacoins {
+		switch {
+		case sco.Value.IsZero():
+			jc.Error(fmt.Errorf("siacoin output %d has zero value", i), http.StatusBadRequest)
+			return
+		case sco.Address == types.VoidAddress:
+			jc.Error(fmt.Errorf("siacoin output %d has void address", i), http.StatusBadRequest)
+			return
+		}
+		siacoinInput = siacoinInput.Add(sco.Value)
+	}
+
+	var siafundInput uint64
+	for i, sfo := range wcr.Siafunds {
+		switch {
+		case sfo.Value == 0:
+			jc.Error(fmt.Errorf("siafund output %d has zero value", i), http.StatusBadRequest)
+			return
+		case sfo.Address == types.VoidAddress:
+			jc.Error(fmt.Errorf("siafund output %d has void address", i), http.StatusBadRequest)
+			return
+		}
+		siafundInput += sfo.Value
+	}
+
+	if siacoinInput.IsZero() && siafundInput == 0 {
+		jc.Error(errors.New("no inputs provided"), http.StatusBadRequest)
+	}
+
+	fee := s.cm.RecommendedFee().Mul64(2000) // use a const for simplicity
+
+	var sent bool
+	var locked []types.Hash256
+	defer func() {
+		if sent {
+			return
+		}
+		s.wm.Release(locked)
+	}()
+
+	sces, basis, siacoinChange, err := s.wm.SelectSiacoinElements(walletID, siacoinInput.Add(fee), false)
```

Comment:
`useUnconfirmed` is probably something we want to expose in the request object?

Reply:
I don't want to expose it here to keep it simple